Tests fail with bottle.py 0.12.17

## What did you try to do?
Run `nosetests ./tests/getinfo_test.py`

## What happened?
With the latest bottle.py (0.12.17), tests fail with:
```
(venv-new-bottle) $ nosetests ./tests/getinfo_test.py
.SFSF.FF...
======================================================================
FAIL: test_getinfo_content_type_invalid_utf8_python3 (tests.getinfo_test.GetinfoTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/yen/tmp/pycurl/tests/util.py", line 91, in decorated
    return fn(*args, **kwargs)
  File "/home/yen/tmp/pycurl/tests/getinfo_test.py", line 117, in test_getinfo_content_type_invalid_utf8_python3
    self.fail('Should have raised')
AssertionError: Should have raised

======================================================================
FAIL: test_getinfo_cookie_invalid_utf8_python3 (tests.getinfo_test.GetinfoTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/yen/tmp/pycurl/tests/util.py", line 91, in decorated
    return fn(*args, **kwargs)
  File "/home/yen/tmp/pycurl/tests/getinfo_test.py", line 88, in test_getinfo_cookie_invalid_utf8_python3
    self.fail('Should have raised')
AssertionError: Should have raised

======================================================================
FAIL: test_getinfo_raw_content_type_invalid_utf8 (tests.getinfo_test.GetinfoTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/yen/tmp/pycurl/tests/getinfo_test.py", line 124, in test_getinfo_raw_content_type_invalid_utf8
    self.assertEqual(expected, self.curl.getinfo_raw(pycurl.CONTENT_TYPE))
AssertionError: b'\xb3\xd2\xda\xcd\xd7' != b'\xc2\xb3\xc3\x92\xc3\x9a\xc3\x8d\xc3\x97'

======================================================================
FAIL: test_getinfo_raw_cookie_invalid_utf8 (tests.getinfo_test.GetinfoTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/yen/tmp/pycurl/tests/getinfo_test.py", line 96, in test_getinfo_raw_cookie_invalid_utf8
    self.assertEqual([expected], self.curl.getinfo_raw(pycurl.INFO_COOKIELIST))
AssertionError: Lists differ: [b'12[26 chars]0\t\xb3\xd2\xda\xcd\xd7\t%96%A6g%9Ay%B0%A5g%A7tm%7C%95%9A'] != [b'12[26 chars]0\t\xc2\xb3\xc3\x92\xc3
\x9a\xc3\x8d\xc3\x97\t%[28 chars]%9A']

First differing element 0:
b'127[25 chars]0\t\xb3\xd2\xda\xcd\xd7\t%96%A6g%9Ay%B0%A5g%A7tm%7C%95%9A'
b'127[25 chars]0\t\xc2\xb3\xc3\x92\xc3\x9a\xc3\x8d\xc3\x97\t%[27 chars]5%9A'

- [b'127.0.0.1\tFALSE\t/\tFALSE\t0\t\xb3\xd2\xda\xcd\xd7\t%96%A6g%9Ay%B0%A5g%A'
-  b'7tm%7C%95%9A']
+ [b'127.0.0.1\tFALSE\t/\tFALSE\t0\t\xc2\xb3\xc3\x92\xc3\x9a\xc3\x8d\xc3\x97\t%96'
+  b'%A6g%9Ay%B0%A5g%A7tm%7C%95%9A']

----------------------------------------------------------------------
Ran 11 tests in 0.117s

FAILED (SKIP=2, failures=4)
```
If I change the first line of requirements-dev.txt to
```
bottle==0.12.16
```
Tests are fine:
```
(venv-old-bottle) $ nosetests ./tests/getinfo_test.py
.S.S.......
----------------------------------------------------------------------
Ran 11 tests in 0.128s

OK (SKIP=2)
```
## What did you expect to happen?
The test suite is compatible with the latest bottle.py.

## What is the PycURL version?
PycURL/7.43.0.2 libcurl/7.65.1 OpenSSL/1.1.1c zlib/1.2.11 libidn2/2.2.0 libpsl/0.21.0 (+libidn2/2.2.0) libssh2/1.8.2 nghttp2/1.39.1

## What is your Python version?
Python 3.7.3

## What is your operating system and its version?
Arch Linux latest

## Other versions

## Is this the most recent PycURL release?
It's actually git-master (9d1fc7fac46b0749477a8ec06b1eb874d8f7125e).

## Did you check libcurl behavior?
I think it's a pycurl issue

--- a/requirements-dev.txt
+++ b/requirements-dev.txt
@@ -1,4 +1,6 @@
-bottle
+# bottle 0.12.17 changed behavior
+# https://github.com/pycurl/pycurl/issues/573
+bottle==0.12.16
 # nose 1.3.1 is broken on python 3:
 # https://github.com/nose-devs/nose/issues/780
 nose>=1.3.2

